IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/WebElement.cs

```diff
@@ -0,0 +1,235 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IntelliTect.TestTools.Selenate
+{
+    public class WebElement : IWebElement
+    {
+        public WebElement(By by, IWebDriver driver)
+        {
+            By = by;
+            _Driver = driver;
+        }
+
+        public WebElement(IWebElement wrappedElement, By by, IWebDriver driver)
+        {
+            WrappedElement = wrappedElement;
+            By = by;
+            _Driver = driver;
+        }
+
+        public By By { get; }
+        public bool Initialized => WrappedElement != null;
+
+        // Finish wrapping these in RetryAction
+        public string TagName => WrappedElement.TagName;
+        public string Text => RetryAction("text").GetAwaiter().GetResult();
+        public bool Enabled => Convert.ToBoolean(RetryAction("enabled").GetAwaiter().GetResult());
+        public bool Displayed => Convert.ToBoolean(RetryAction("displayed").GetAwaiter().GetResult());
+        public bool Selected => Convert.ToBoolean(RetryAction("selected").GetAwaiter().GetResult());
+        public Point Location => WrappedElement.Location;
+        public Size Size => WrappedElement.Size;
+
+        public WebElement FindElement(By by)
+        {
+            if (!Initialized)
+            {
+                WrappedElement = _Driver.FindElement(by);
+            }
+
+            WebDriverWait wait = new WebDriverWait(_Driver, TimeSpan.FromSeconds(5));
+            try
+            {
+                wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+            }
+            catch (WebDriverTimeoutException)
+            {
+                return new WebElement(by, _Driver);
+            }
+
+            return new WebElement(WrappedElement.FindElement(by), by, _Driver);
+        }
+
+        public ReadOnlyCollection<WebElement> FindElements(By by)
+        {
+            if (!Initialized)
+            {
+                WrappedElement = _Driver.FindElement(by);
+            }
+
+            WebDriverWait wait = new WebDriverWait(_Driver, TimeSpan.FromSeconds(5));
+            try
+            {
+                wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+            }
+            catch (WebDriverTimeoutException)
+            {
+                return new ReadOnlyCollection<WebElement>(new List<WebElement>());
+            }
+
+            return
+                new ReadOnlyCollection<WebElement>(
+                    WrappedElement.FindElements(by)
+                        .Select(webElement =>
+                        new WebElement(WrappedElement.FindElement(by), by, _Driver))
+                        .ToList());
+        }
+
+        public void Clear()
```

Comment:
Why is this a `void` return method. I would expect that the async refactor would cause this to be:
`public async Task Clear()`.

Similar comment for the other methods in this file.